File: leaf-system/src/main/java/com/leaf/system/service/LeafDictService.java
package com.leaf.system.service;


import com.leaf.system.domain.LeafDict;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;


/**
 * 字典表业务层
 *
 * @author jhlz
 * @since 2023-05-07 14:29:03
 */
public interface LeafDictService {

    /**
     * 分页列表
     *
     * @param leafDict 条件参数
     * @param page     分页参数
     * @return 符合条件的分页数据
     */
    Page<LeafDict> listPage(LeafDict leafDict, Pageable page);

    /**
     * 根据 ID 获取详情
     *
     * @param id id
     * @return 详情对象
     */
    LeafDict findById(Long id);

    /**
     * 添加数据
     *
     * @param leafDict 添加数据内容
     */
    LeafDict save(LeafDict leafDict);

    /**
     * 更新数据
     *
     * @param leafDict 更新的数据内容
     */
    void update(LeafDict leafDict);

    /**
     * 删除数据
     *
     * @param ids 删除数据的 id 集合
     */
    void remove(List<Long> ids);
}

File: src/main/kotlin/com/jhlz/service/LeafDictService.kt
package com.jhlz.service

import com.jhlz.domain.LeafDict
import com.jhlz.domain.fetchBy
import com.jhlz.record.PageQuery
import jakarta.inject.Singleton
import org.babyfish.jimmer.Page
import org.babyfish.jimmer.sql.kt.KSqlClient

/**
 * @author jhlz
 * @version x.x.x
 */
@Singleton
class LeafDictService(val sqlClient: KSqlClient) {
    fun page(pageQuery: PageQuery): Page<LeafDict> {
        return sqlClient.createQuery(LeafDict::class) {
            select(
                table.fetchBy {
                    allScalarFields()
                    creator { username() }
                }
            )
        }.fetchPage(pageQuery.pageNum - 1, pageQuery.pageSize)
    }

    fun info(id: Long): LeafDict {
        return sqlClient.findById(LeafDict::class, id)!!
    }

    fun add(dict: LeafDict) {
        sqlClient.entities.save(dict)
        throw Exception("test")

    }

    fun update(dict: LeafDict) {
        sqlClient.entities.save(dict)
    }

    fun delete(ids: List<Long>) {
        sqlClient.deleteByIds(LeafDict::class, ids)
    }
}